docs/upgrade/concept.md

```diff
@@ -0,0 +1,57 @@
+---
+layout: default
+title: Overall concept
+parent: Firmware Upgrade
+nav_order: 810
+---
+## Overall Concept for Firmware Upgrades
+
+For any modern device with evolving features, especially in a security-critical environment, regular updates are an absolute must. Providing new features, improvements and timely security-patches is a key focus of our project.
+
+In terms of upgrading a system, there are several approaches:
+
+* **script-based, no modularization**: upgrading the system with scripts, without modularization of individual components, is a lost cause as differences and issues accumulate over time. 
+
+* **modular updates**: a more reliable approach is to encapsulate components in individual modules, e.g. using Docker. Every module can be updated to a defined state. Issues can arise when different versions of modules are not working well together, or when the base operating system needs updating.
+
+* **disk image update**: updates overwrite the whole disk partition, usually containing the operating system and applications. This results in a very reliable, cleanly defined state. The whole system can be tested in-depth beforehand, but this low-level method adds complexity to the upgrade process.
+
+### Our approach: atomic disk image updates
+We are working towards the goal to build the BitBox Base as an appliance with firmware, not as a small Linux server. This is why we decided to implement a full disk image update process, using the [Mender](https://mender.io/) open-source solution. 
+
+This solution has the following features:
+* **enterprise-grade**: despite being open-source and free to use, Mender is a professional solution with its own Deployment Management server and a robust client update daemon.
+* **full disk image**: the whole operating system including all applications are updated together, resulting in a clearly defined state of the system.
+* **atomic**: the update process is atomic, as it either succeeds in full (including custom application tests), or not at all. In case of failure, a full rollback is performed.
+* **on-demand**: the BitBox App will prompt you when an update is available, but you're not forced to upgrade. 
```

Comment:
Regarding `you're not forced to upgrade`, will this always be the case? I.e if we knew there is a critical security vulnerability in a current release putting the user's funds at risk, would we never consider requiring the upgrade?

Reply:
I'd assume so? Forcefully upgrading the Base would be an additional "feature" that we did not talk about yet. But I cannot really see a scenario, where "forcing" the user, instead of "nagging" him very much would solve the problem.

Maybe it's more about a background update process, opt-in or -out to install critical security patches automatically and directly, without going through the app? Needs to be discussed.

Reply:
> Needs to be discussed.

Sure, sounds good. So for the purposes of these docs, perhaps it's best to just leave in the part stating that `the BitBox App will prompt you when an update is available`, but omit the part saying `, but you're not forced to upgrade` until we're sure that will remain the case?

Reply:
The feature that "needs to be discussed" is opt-in or -out, so no "forcing" here. I don't see any scenario where we would force an updated if the user does not wish so.

Reply:
Okay.